File: solutions/0121-best-time-to-buy-and-sell-stock/solution.go
func maxProfit(prices []int) int {
    maxProfit := 0
    minPrice := int ((^uint(0)) >> 1)
    for _, p := range prices {
        if p < minPrice {
            minPrice = p
            // Deduct with other prices
            continue
        }
        if m := (p - minPrice); m > maxProfit {
            maxProfit = m
        }
    }
    return maxProfit
}

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m = 0, min = INT_MAX;
        if (prices.empty()) return m;
        for (auto& p: prices) {
            if (p < min) {
                min = p;
                continue;
            }
            if ((p - min) > m) {
                m = p - min;
            }
        }
        return m;
    }
};
